include/nbl/system/ISystem.h

```diff
@@ -266,33 +274,36 @@ class ISystem final : public core::IReferenceCounted
     }
 
     //! Warning: blocking call
-    core::smart_refctd_ptr<IFileArchive> createFileArchive(const std::filesystem::path& filename)
+    core::smart_refctd_ptr<IFileArchive> createFileArchive(const std::filesystem::path& filename, const std::string_view& password = "")
     {
         future_t<core::smart_refctd_ptr<IFile>> future;
         if (!createFile(future, filename, IFile::ECF_READ))
             return nullptr;
 
         auto file = std::move(future.get());
 
-        return createFileArchive(file.get());
+        return createFileArchive(std::move(file), password);
     }
-    core::smart_refctd_ptr<IFileArchive> createFileArchive(IFile* file)
+    core::smart_refctd_ptr<IFileArchive> createFileArchive(core::smart_refctd_ptr<IFile>&& file, const std::string_view& password = "")
     {
         if (file->getFlags() & IFile::ECF_READ == 0)
             return nullptr;
 
-        // @sadiuk implement in manner similar to IAssetManager::getAsset
-
+        auto ext = system::extension_wo_dot(file->getFileName());
+        auto loaders = m_loaders.perFileExt.findRange(ext);
+        for (auto& loader : loaders)
+        {
+            return loader.second->createArchive(std::move(file));
+        }
         return nullptr;
     }
 
-    void mount(core::smart_refctd_ptr<IFileArchive>&& archive, const system::path& pathAlias)
+    void mount(core::smart_refctd_ptr<IFileArchive>&& archive, const system::path& pathAlias = "")
     {
-        const auto& files = archive->getFiles();
-        for (auto& archFile : files)
+        m_cachedArchiveFiles.insert(archive->asFile()->getFileName(), std::move(archive));
+        if (!pathAlias.empty())
```

Comment:
I wouldn't keep a map of `m_cachedPathAliases` just insert into `m_cachedArchiveFiles` with a `pathAlias` instead of `archive->asFile()->getFileName()` if an alias is given